include/dlaf/communication/init.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+/// @file
+
+#include "mpi.h"
+
+#include <cstdio>
+
+namespace dlaf {
+namespace comm {
+
+/// A RAII type to simplify MPI's init/finalize
+struct InitMPI {
+
+  InitMPI(int argc, char** argv, int thd_required) {
+    int thd_provided;
+    MPI_Init_thread(&argc, &argv, thd_required, &thd_provided);
+
+    if (thd_required != thd_provided) {
+      std::printf("Provided MPI threading model does not match the required one.\n");
+      MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+  }
+
+  ~InitMPI() {
+    MPI_Finalize();
+  }
+};
```

Comment:
I think it is a nice initiative that can be part of https://github.com/eth-cscs/DLA-Future/issues/100